src/integrationtest/java/com.microsoft.aad.msal4j/AuthorizationCodeIT.java

```diff
@@ -30,6 +31,14 @@ public void acquireTokenWithAuthorizationCode_ManagedUser(String environment){
         assertAcquireTokenAAD(user);
     }
 
+    @Test(dataProvider = "environments", dataProviderClass = EnvironmentsProvider.class)
+    public void acquireTokenWithAuthorizationCode_ManagedUserWithClaims(String environment){
+        cfg = new Config(environment);
+
+        User user = labUserProvider.getDefaultUser(cfg.azureEnvironment);
+        assertAcquireTokenAADWithClaims(user);
```

Comment:
I think you should be able to refactor some of the code that you added, as there is a bit of duplication. For example, consider using `assertAcquireTokenAAD`, and passing an extra parameter for claims. You should then be able to use of all of the same methods, and in `buildAuthenticationCodeURL` you could check if claims were passed in, and if so then add them to the client application.

Reply:
I refactored it to remove most of the duplication by adding a new parameter to `assertAcquireTokenAAD` to hold the auth request URL parameters, which is just passed along until `acquireAuthorizationCodeAutomated`. There, it's used to either build the URL using an existing method (with no optional params), or build the URL with a new method that can handle optional params. For now it just looks for claims+capabilities, but it can be extended to handle tests of other optional params as needed.